### PR TITLE
CDAP-8007 Use a separate thread for scanning TMS Topic Metadata Table in TMS HBase coprocessors 

### PR DESCRIPTION
**Pull request by gokulavasan**
JIRA : CDAP-8007

Build : http://builds.cask.co/browse/CDAP-RUT449

TODO: Port changes to other compat modules. Current commits change only 0.98 and 1.1 compat modules. Will port the changes to other modules after review.

### CODE REVIEW

**Comment by gokulavasan** [comment]
@chtyim Updated PR to remove the extra class in every compat module (moved that logic to TopicMetadataCache). This was possible after I moved the CConfigurationReader, CConfigurationTable to cdap-hbase-compat-base.

**Comment by gokulavasan** [comment]
@chtyim Addressed your comment by shutting down the refresh thread in TMS table coprocessors in the last commit.

**Comment by chtyim** [comment]
LGTM

**Comment by gokulavasan** [comment]
Build is 🍏

**Comment by anew** [comment]
please update all the copyright headers

File: cdap-hbase-compat-0.98/src/main/java/co/cask/cdap/data2/transaction/messaging/coprocessor/hbase98/MessageTableRegionObserver.java [hunk 1]
```diff
@@ -187,31 +126,31 @@ public InternalScanner preCompactScannerOpen(ObserverContext<RegionCoprocessorEn
...
-      return super.preCompactScannerOpen(c, store, scanners, scanType, earliestPutTs, s, request);
-    }
-
+    TopicMetadataCache metadataCache = getTopicMetadataCache(c.getEnvironment());
     LOG.info("preCompact, filter using MessageDataFilter");
     TransactionVisibilityState txVisibilityState = txStateCache.getLatestState();
     Scan scan = new Scan();
     scan.setFilter(new MessageDataFilter(c.getEnvironment(), System.currentTimeMillis(),
-                   prefixLength, cache, txVisibilityState));
+                   prefixLength, metadataCache, txVisibilityState));
```
Comment (anew) [inline cdap-hbase-compat-0.98/src/main/java/co/cask/cdap/data2/transaction/messaging/coprocessor/hbase98/MessageTableRegionObserver.java]: fix indentation here?

File: cdap-hbase-compat-0.98/src/main/java/co/cask/cdap/data2/transaction/messaging/coprocessor/hbase98/MessageTableRegionObserver.java [hunk 2]
```diff
@@ -187,31 +126,31 @@ public InternalScanner preCompactScannerOpen(ObserverContext<RegionCoprocessorEn
...
+  }
+
+  private TopicMetadataCache createTopicMetadataCache(RegionCoprocessorEnvironment env) {
+    return new TopicMetadataCache(env, cConfReader, nameConverter, hbaseNamespacePrefix, metadataTableNamespace,
+                                  new DefaultScanBuilder());
+  }
+
   private Supplier<TransactionStateCache> getTransactionStateCacheSupplier(String tablePrefix, Configuration conf) {
-    String sysConfigTablePrefix = new HTable98NameConverter().getSysConfigTablePrefix(tablePrefix);
+    String sysConfigTablePrefix = nameConverter.getSysConfigTablePrefix(tablePrefix);
```
Comment (anew) [inline cdap-hbase-compat-0.98/src/main/java/co/cask/cdap/data2/transaction/messaging/coprocessor/hbase98/MessageTableRegionObserver.java:153]: looks like this change did not get ported over to the other modules

**Comment by anew** [comment]
@gokulavasan only three comments then LGTM

**Comment by gokulavasan** [comment]
@anew Thanks for the review! Please take a look at the last commit.

**Review by anew — APPROVED** [review]
LGTM

**Pull request merged**